src/components/ApplicationInsightService.tsx

```diff
@@ -0,0 +1,29 @@
+import { ApplicationInsights, ITelemetryItem } from '@microsoft/applicationinsights-web';
+import { ReactPlugin } from '@microsoft/applicationinsights-react-js';
+
+interface ApplicationInsightProps {
+  connectionString: string;
+  customEventProperties?: {};
+  customEventName: string;
+}
+
+export const ApplicationInsightService = (props: ApplicationInsightProps) => {
```

Comment:
I think this ApplicationInsightsService should be a singleton. But currently it is being initialised as many times as many events are being sent

Reply:
Added changes. Thanks